### DISCUSSION TITLE
Fade out CatchComboDisplay when hide HUD

### DISCUSSION THREAD

**Discussion by cdwcgt** [ppy/osu#20217] [Ideas] [answered]
This is consistent with the behavior of stable

before
![image](https://user-images.githubusercontent.com/34775378/189472174-a9c9aabf-9b68-4914-9b36-ca25a9c2234c.png)

after
![image](https://user-images.githubusercontent.com/34775378/189473350-379b0cee-cc3a-4f34-be13-cc4b451a1777.png)

But how to say I would like to have it as an option instead of following the HUD
because I just want remove combo display **here** or not all hud

**Comment by peppy** [comment]
It would be best if it was implemented with a character tracking toggle or similar.

The ability to hide/show show probably be deferred for now, as we have plans on different display modes for all hud components.

**Reply by cdwcgt** [reply]
I made an implementation, based on [CatchComboDisplay.cs](https://github.com/ppy/osu/blob/master/osu.Game.Rulesets.Catch/UI/CatchComboDisplay.cs), but this implementation I don't think is particularly good because I can't get the `ShowHUD` property from the HUDOverlay so I wrote the display judgment again(Requires DrawableRuleset and OsuConfigManager to provide replayLoaded and HUDVisibilityMode)

May need HUDOverlay to be loaded first and other drawable can easy access to dependencies?

**Reply by peppy** [reply]
The catch combo is in a very awkward place. You're welcome to pull request what you have and we can discuss there. Maybe fine as a temporary solution.

**Comment by cdwcgt** [comment, marked as answer]
https://github.com/ppy/osu/pull/20266
